pin_test.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/datadog/orchestrion/internal/version"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPin(t *testing.T) {
+	tmp := t.TempDir()
+	if cwd, err := os.Getwd(); err == nil {
+		defer require.NoError(t, os.Chdir(cwd))
+	}
+
+	require.NoError(t, scaffold(tmp))
+	require.NoError(t, os.Chdir(tmp))
+	requiredVersionError = errors.New("test")
+	autoPinOrchestrion()
+	require.NoError(t, requiredVersionError)
+
+	require.FileExists(t, filepath.Join(tmp, orchestrionToolGo))
+	require.FileExists(t, filepath.Join(tmp, "go.sum"))
+
+	data, err := os.ReadFile(filepath.Join(tmp, "go.mod"))
+	require.NoError(t, err)
+	require.Contains(t, string(data), fmt.Sprintf(`github.com/datadog/orchestrion %s`, version.Tag))
```

Comment:
`go list` might be a good double-validation that the go.mod file isn't broken

Reply:
That's not a bad idea :)